pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -255,7 +257,10 @@ def circuit():
                 tape = tape_inside
 
     for op in tape.operations:
-        op._wires = Wires([inverse_wires_map[wire] for wire in op.wires.tolist()])
+        if isinstance(op, qml.ops.arithmetic.Adjoint):
+            op.base._wires = Wires([inverse_wires_map[wire] for wire in op.wires.tolist()])
```

Comment:
Why is this necessary? I thought the base and adjoint wires are the same?

Reply:
`Adjoint` doesn't have an `_wires` attribute.  We could also solve this problem by adding an official setter for wires, so that wires can be updated without hacking into a private attribute.

Reply:
Maybe not so important for now, but one day we should revise wires for ops...don't think it is a lot of work!

Reply:
I had concerns before, but I agree that a wires setter (as an exception of a mutable structure) would be nice!